test/controllers/api/v2/locations_controller_test.rb

```diff
@@ -335,4 +335,63 @@ def add_locations
     assert_response :success
     assert_equal 1, location.reload.location_parameters.count
   end
+
+  test "should create with comma separated name" do
+    name = "#{RFauxFactory.gen_alpha}, #{RFauxFactory.gen_alpha}"
+    post :create, params: { :location => { :name => name } }
+    assert_response :success, "creation with name #{name} failed with code #{response.code}"
+    show_response = JSON.parse(@response.body)
+    refute_empty show_response
+  end
+
+  test "should create with description" do
+    name = RFauxFactory.gen_alpha
+    description = RFauxFactory.gen_alpha
+    post :create, params: { :location => { :name => name, :description => description } }
+    assert_response :success, "creation with name #{name} and description #{description} failed with code #{response.code}"
+    show_response = JSON.parse(@response.body)
+    refute_empty show_response
+  end
+
+  test "should not create with same name" do
+    name = RFauxFactory.gen_alphanumeric
+    post :create, params: { :location => { :name => name } }
```

Comment:
these two lines could be replaced by simple `FactoryBot.create(:location, :name => name)`, that will ensure such object already exists and we don't need to go through the whole controller, since this is already tested by `test "should create with comma separated name"`. The subject of this test is the second post, that should fail on validation problem.

Another reason is that in future, all controller tests should be limited to single API call.